### PR TITLE
add updated licensing guidelines to docs

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <admiller@redhat.com>

##### SUMMARY
It was decided during the Ansible Core Developer Community Meeting
on 2018-08-14 that GPLv3 is acceptable for module_utils in specific
scenarios.

https://github.com/ansible/community/issues/329#issuecomment-413005278

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

docs

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (docs/licensing ee770890d0) last updated 2018/08/14 16:50:49 (GMT -500)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/admiller/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/admiller/src/dev/ansible/lib/ansible
  executable location = /home/admiller/src/dev/ansible/bin/ansible
  python version = 3.6.6 (default, Jul 19 2018, 14:25:17) [GCC 8.1.1 20180712 (Red Hat 8.1.1-5)]
```

### CODE REVIEW

File: docs/docsite/rst/dev_guide/shared_snippets/Licensing.txt
Comment (abadger) [inline docs/docsite/rst/dev_guide/shared_snippets/Licensing.txt:3]: GPLv3 or later.  (in case there's ever a GPLv3.1, GPLv4, etc)

File: docs/docsite/rst/dev_guide/shared_snippets/Licensing.txt
Comment (abadger) [inline docs/docsite/rst/dev_guide/shared_snippets/Licensing.txt:2]: It's all files except for module_utils.

**Review by acozine — CHANGES_REQUESTED** [review]
+1 to the change, but please don't merge it via this PR - I've got a massive re-write in progress on the Developer Guide - see #44098. I'll assign this PR to myself and incorporate it either as part of the re-write, or after the re-write is merged.

**Comment by maxamillion** [comment]
@abadger should I specify 2-clause BSD or does it matter?

**Comment by abadger** [comment]
It does matter but probably best to point to our boilerplate and the
copies of the license in our repo

**Comment by abadger** [comment]
* license copies: https://github.com/ansible/ansible/tree/devel/licenses
* boilerplate: https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/dev_guide/developing_modules_documenting.rst#id1

Perhaps the snippet should point to the  developing_modules_documenting.rst section on Copyright and we add the bulk of this information there. (It will also be good to add the one-liner for BSD licensed code there.  (The one-liner for BSD can be taken from basic.py: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L3  

```# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)```

**Comment by acozine** [comment]
Closing in favor of #45574 to incorporate changes from #45349.

**Pull request closed without merging**
